app/assets/javascripts/admin/addon/components/color-palette-editor.gjs

```diff
@@ -78,28 +54,58 @@ const Picker = class extends Component {
     const color = event.target.value.replace("#", "");
     if (this.args.showDark) {
       this.args.onDarkChange(color);
-      this.args.color.darkValue = color;
     } else {
       this.args.onLightChange(color);
-      this.args.color.lightValue = color;
     }
   }
 
   @action
   onTextChange(event) {
-    const color = event.target.value;
+    let color = event.target.value;
+
+    if (!this.isValidHex(color)) {
+      event.preventDefault();
+      return;
+    }
+
+    color = this.ensureSixDigitsHex(color);
     if (this.args.showDark) {
       this.args.onDarkChange(color);
-      this.args.color.darkValue = color;
     } else {
       this.args.onLightChange(color);
-      this.args.color.lightValue = color;
     }
   }
 
   @action
   onTextKeypress(event) {
-    const color = event.target.value + event.key;
+    const currentValue = event.target.value;
+
+    if (event.keyCode === 13) {
+      event.preventDefault();
+
+      if (currentValue.length !== 6 && currentValue.length !== 3) {
+        this.toasts.error({
+          data: {
+            message: i18n(
+              "admin.config_areas.color_palettes.invalid_color_length"
+            ),
+          },
+        });
```

Comment:
Personally, I feel like using a `toast` here is not the right UI pattern. Normally, input validation errors are shown right below the input so that it is clear which input the error is shown for.

Reply:
Just to make it clear, I don't think this should be a blocking comment but something worth thinking about.

cc @ellaestigoy